### ISSUE TITLE
How can I start/stop my rig through automation in Home Assistant?

### ISSUE THREAD

**Issue by old-square-eyes** [RomRider/ha_nicehash#25]
Is it a device, service or something else in the actions? I have a trigger. I can't figure out what to fire. Sorry, a bit new to Home Assistant.

**Comment by old-square-eyes** [comment]
Or is it this (a service)

<img width="970" alt="image" src="https://user-images.githubusercontent.com/25760491/222949258-ac7e8b48-361c-403d-ac8c-bcd0fad9f034.png">

**Comment by old-square-eyes** [comment]
Bump

**Comment by jseppeli** [comment]
Also interested to know

**Comment by old-square-eyes** [comment]
I was able to get automation bumping the power mode when my solar is exporting excess energy to the grid. I have one in reverse too for when there's no longer any surplus, setting power mode to LOW. But as far as I can tell there is no complete START/STOP

![image](https://user-images.githubusercontent.com/25760491/229258660-422336d7-982a-40ce-a4ed-235d742525bd.png)

**Comment by old-square-eyes** [comment]
I got start/stop to work with Action > Service > Switch "Turn On", then added the entities...

Issue is though that NH on my PC seems to lock up and become unresponsive after an API start, followed by an API stop. I can't even start/stop in the desktop app. And I couldn't exit it at one point with the app icon stuck in the system tray. I had to reboot my PC. The weird state is the desktop app says mining with zero profitability. Online monitor says "Status Unknown".

![image](https://user-images.githubusercontent.com/25760491/229261560-fd12a1a6-8da6-4640-aec2-374626a4feef.png)

**Comment by jseppeli** [comment]
I can't find the switches in list of entities. Am I possibly missing something important?

**Comment by old-square-eyes** [comment]
> I can't find the switches in list of entities. Am I possibly missing something important?

In automation add a Service. In services add a Switch. In Switch add the Nicehash Entity.

**Comment by jseppeli** [comment]
> > I can't find the switches in list of entities. Am I possibly missing something important?
> 
> In automation add a Service. In services add a Switch. In Switch add the Nicehash Entity.

Like I tried telling above, I can't find the Nicehash switches as Entities.
Here is a complete list of all switch entities that can be selected and they are not either shown in Devicese _& Servicec entity list
![image](https://user-images.githubusercontent.com/128599472/229342806-312599f1-9e09-478a-9721-8b7694bbe7dd.png)

**Comment by old-square-eyes** [comment]
@jseppeli have you installed this integration per the instructions via HACS?

**Comment by jseppeli** [comment]
@old-square-eyes yes and I have even checked that API has the same permissions as in the instructions

**Comment by eavogels** [comment]
> 
Hi. Is there a follow up here? I'm exactly at this pojnt as well. Cannot find the switches.  Did you get it to work?

**Comment by jseppeli** [comment]
> > 
> 
> Hi. Is there a follow up here? I'm exactly at this pojnt as well. Cannot find the switches. Did you get it to work?

No, never got it working

**Comment by julianxv** [comment]
@jseppeli @eavogels 

To get the Power Switches you've to do a change at the switch.py File in custom_components/nicehash.

Changes that have to be done are listed here:

https://github.com/RomRider/ha_nicehash/compare/master...RoaringSon:ha_nicehash:master
